numpy/lib/tests/test_shape_base.py

```diff
@@ -403,6 +403,13 @@ def test_integer_split_2D_default(self):
         assert_(a.dtype.type is res[-1].dtype.type)
         # perhaps should check higher dimensions
 
+    def test_integer_split_2D_rows_greater_max_int32(self):
+        a = np.broadcast_to([0], (1 << 32, 2))
+        res = array_split(a, 4)
+        chunk = np.broadcast_to([0], (1 << 30, 2))
+        tgt = [chunk] * 4
+        compare_results(res, tgt)
```

Comment:
This will not work, because it will actually compare the full arrays (and by that even create a full boolean array fo reach of the chunks even). It will be slow and use a lot of memory. Please replace it with a test that only checks the shape.

Reply:
@seberg Done!